src/pages/QuizPage.tsx

```diff
@@ -28,8 +56,15 @@ export const QuizPage = () => {
         <InputComponent type="radio" className={'radio-btn'} name={'answer'} id="answer3" labelText={'5'} />
       </section>
       <section className="buttons-wrapper">
-        <ButtonComponent className={'show-btn'} text={'Quit'} />
-        <ButtonComponent className={'quit-btn'} text={'Next'} />
+        <ButtonComponent
+          className={'quit-btn'}
+          text={'Quit'}
+          onClick={() => {
+            setModalIsOpen(true);
+          }}
+        />
+        <ButtonComponent className={'quit-btn'} text={'Next'} onClick={nextAction} />
+        <ModalComponent isOpen={modalIsOpen} children={modalContent} />
```

Comment:
please remove `children`
and put the content between `ModalComponent` tags
such as
```tsx
<ModalComponent>
 {modalContent}
</ModalComponent>
```